### PR TITLE
MSONAR-204 "SonarQube version" shouldn't be displayed when communicating with SonarCloud

### PR DESCRIPTION
**Pull request by leveretka**

### CODE REVIEW

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java [hunk 1]
```diff
@@ -38,12 +40,16 @@
...
+import static org.sonarsource.scanner.maven.bootstrap.MavenProjectConverter.getPropertyByKey;
+
 /**
  * Configure properties and bootstrap using SonarQube scanner API
  */
 public class ScannerBootstrapper {
 
   static final String UNSUPPORTED_BELOW_SONARQUBE_56_MESSAGE = "With SonarQube server prior to 5.6, use sonar-maven-plugin <= 3.3";
+  private static final String SONAR_HOST_URL_KEY = "sonar.host.url";
+  private static final String SONAR_HOST_URL = "https://sonarcloud.io";
```
Comment (agigleux) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java:52]: What about using SONARCLOUD in the name of the constant to be more explicit when reading the code where this constant is used?

```suggestion
  private static final String SONARCLOUD_HOST_URL = "https://sonarcloud.io";
```

**Review by agigleux — CHANGES_REQUESTED** [review]

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java [hunk 2]
```diff
@@ -65,7 +71,12 @@ public void execute() throws MojoExecutionException {
       scanner.start();
       serverVersion = scanner.serverVersion();
 
-      checkSQVersion();
+      if (isSonarCloudUsed()) {
+        log.info("Communicating with SonarCloud");
+      } else {
+        checkSQVersion();
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java:77]: I think this is a good opportunity to remove the logging from `checkSQVersion` and restrict to checking that we can talk with the version.
```suggestion
        checkSQVersion();
        if (serverVersion != null) {
          log.info("Communicating with SonarQube Server " + serverVersion);
        }
```

```java
private void checkSQVersion() {
  if (isVersionPriorTo("5.6")) {
    throw new UnsupportedOperationException(UNSUPPORTED_BELOW_SONARQUBE_56_MESSAGE);
  }
}
```

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java [hunk 3]
```diff
@@ -38,12 +40,16 @@
...
 
+import static org.sonarsource.scanner.maven.bootstrap.MavenProjectConverter.getPropertyByKey;
+
 /**
  * Configure properties and bootstrap using SonarQube scanner API
  */
 public class ScannerBootstrapper {
 
   static final String UNSUPPORTED_BELOW_SONARQUBE_56_MESSAGE = "With SonarQube server prior to 5.6, use sonar-maven-plugin <= 3.3";
+  private static final String SONAR_HOST_URL_KEY = "sonar.host.url";
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java:51]: We can replace this one with `ScannerProperties.HOST_URL` that is already used elsewhere in the code

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
The changes look good but I think we can split out some of the work and reuse some existing properties

**Review by agigleux — APPROVED** [review]

**Pull request merged**